ScrapPackedTests/TestCliApp.cs

```diff
@@ -289,114 +246,304 @@ public void TestRunExtract()
+            File.Copy(@"TestData\empty.packed", @"TestResults\TestOutputPackedFail\packedFile.packed", true);
+
+            // Access denied
+            Directory.CreateDirectory(@"TestResults\TestOutputPackedFail\filenameWasTaken");
+            CheckRunFail(new[] { "add", "--packedFile", @"TestResults\TestOutputPackedFail\packedFile.packed",
+                "--sourcePath", @"TestData\examplefile1.txt",
+                "--packedPath", "folder/file.txt",
+                "--outputPackedFile", @"TestResults\TestOutputPackedFail\filenameWasTaken" },
+                1, "Access to output file is denied");
+
+            Directory.Delete(@"TestResults\TestOutputPackedFail\filenameWasTaken");
+
+            // check inaccessable output packed
+            var fsFile = new FileStream(@"TestResults\TestOutputPackedFail\packedOutFile.packed", FileMode.OpenOrCreate);
+            try
+            {
+                CheckRunFail(new[] {"add", "--packedFile", @"TestResults\TestOutputPackedFail\packedFile.packed",
+                    "--sourcePath", @"TestData\examplefile1.txt",
+                    "--packedPath", "file.txt",
+                    "--outputPackedFile", @"TestResults\TestOutputPackedFail\packedOutFile.packed" },
+                    1, "Expected output file to be inaccessible");
+            }
+            finally
+            {
+                fsFile.Close();
+            }
+
+            if (Directory.Exists("TestResults"))
+                Directory.Delete("TestResults", true);
+        }
+
+        [TestMethod]
+        public void TestBackup()
+        {
+            Directory.CreateDirectory(@"TestResults\TestBackup\");
+            File.Copy(@"TestData\empty.packed", @"TestResults\TestBackup\packedFile.packed", true);
+            File.Copy(@"TestData\examplefile1.txt", @"TestResults\TestBackup\examplefile1.txt", true);
+
+            CheckRunCompareFile(new[] {"add", "--packedFile", @"TestResults\TestBackup\packedFile.packed",
+                "--sourcePath", @"TestData\examplefile1.txt",
+                "--packedPath", @"file.txt" },
+                @"TestResults\TestBackup\packedFile.packed.bak",
+                @"TestData\empty.packed",
+                "Backup after adding file to new"
+            );
+
+            CheckRunCompareFile(new[] {"add", "--packedFile", @"TestResults\TestBackup\packedFile.packed",
+                "--sourcePath", @"TestData\examplefile1.txt",
+                "--packedPath", @"file.txt" },
+                @"TestResults\TestBackup\packedFile.packed.bak",
+                @"TestResults\TestBackup\packedFile.packed",
+                "Backup after re-adding file"
+            );
+
+            // todo: this test is not working because app even can't make backup. 
+            //       Need to find way to open file/make folder with name "examplefile1.txt" 
+            //       after making backup but before tying to extract.
```

Comment:
comment … not intended to fix now:

ok, this might be where we need to start using mock objects